src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs

```diff
@@ -81,6 +101,31 @@ protected override void Shutdown()
 
         #region Built-in Message Handlers
 
+        protected async Task HandleConfigurationDoneRequest(
+            object args,
+            RequestContext<object> requestContext)
+        {
+            // Ensure that only the second message between launch and
+            // configurationDone requests, actually launches the script.
+            lock (syncLock)
```

Comment:
No problem!  There can be some more subtle behavior with async methods that could require locking but we can discuss that another time ;)